VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -171,6 +172,24 @@ public void onDestroy() {
         EventBus.getDefault().unregister(this);
     }
 
+    @Override
+    public void onRestart() {
+        if (postsSort == DiscussionPostsSort.LAST_ACTIVITY_AT && mSelectedItem != -1) {
+            // Move the last viewed thread to the top
+            discussionPostsAdapter.moveToTop(mSelectedItem);
+            mSelectedItem = -1;
```

Comment:
This can be done directly from the `onItemClick()` callback, thus removing the need to store the click index. Or, if this change is somehow visible through the new Activity transition, then this action can be posted on the event queue (i.e. via a `Handler`) from there.